nibabel/gifti/parse_gifti_fast.py

```diff
@@ -309,64 +321,25 @@ def flush_chardata(self):
             # update the endianness according to the
             # current machine setting
             self.endian = gifti_endian_codes.code[sys.byteorder]
+
         elif self.write_to == 'Label':
             self.label.label = data.strip()
 
     @property
     def pending_data(self):
         " True if there is character data pending for processing "
-        return not self._char_blocks is None
+        return self._char_blocks is not None
 
 
-def parse_gifti_file(fname=None, fptr=None, buffer_size=None):
-    """ Parse gifti file named `fname`, return image
-
-    Parameters
-    ----------
-    fname : str
-        filename of gifti file
-    buffer_size: None or int, optional
-        size of read buffer. None gives default of 35000000 unless on python <
-        2.6, in which case it is read only in the parser.  In that case values
-        other than None cause a ValueError on execution
-
-    Returns
-    -------
-    img : gifti image
-    """
-    assert (fname is not None) + (fptr is not None) == 1, "Specify only fname or fptr, not both"
-
-    if fptr is None:
-        with open(fname, 'rb') as datasource:
-            return parse_gifti_file(fptr=datasource, buffer_size=buffer_size)
-    else:
-        datasource = fptr
+class Outputter(GiftiImageParser):
+    @np.deprecate_with_doc("Use GiftiImageParser instead.")
+    def __init__(self, *args, **kwargs):
+        super(Outputter, self).__init__(*args, **kwargs)
 
-    if buffer_size is None:
-        buffer_sz_val = 35000000
-    else:
-        buffer_sz_val = buffer_size
-
-    parser = ParserCreate()
-    parser.buffer_text = True
-    try:
-        parser.buffer_size = buffer_sz_val
-    except AttributeError:
-        if not buffer_size is None:
-            raise ValueError('Cannot set buffer size for parser')
-    HANDLER_NAMES = ['StartElementHandler',
-                     'EndElementHandler',
-                     'CharacterDataHandler']
-    out = Outputter()
-    for name in HANDLER_NAMES:
-        setattr(parser, name, getattr(out, name))
-    try:
-        parser.ParseFile(datasource)
-    except ExpatError:
-        print('An expat error occured while parsing the  Gifti file.')
-
-    # Reality check for pending data
-    assert out.pending_data is False
-    # update filename
-    out.img.filename = fname
-    return out.img
+    def initialize(self, *args, **kwargs):
+        self.__init__(*args, **kwargs)
```

Comment:
Setting `encoding` and `buffer_size` is new. I figured since `initialize` is deprecated, nobody will call it with those args, nor would those args be non-defaults. I'd prefer to keep the # of lines for deprecated functions minimal.

As a side note, I could remove the arguments from `initialize` and `__init__` to match the original definitions.

LMK what you think.